src/main/java/uk/gov/hmcts/reform/idam/service/DeleteUserService.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.idam.service;
+
+import jakarta.ws.rs.core.Response;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.idam.parameter.ParameterResolver;
+import uk.gov.hmcts.reform.idam.service.remote.RestClient;
+
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+
+@Service
+@Slf4j
+public class DeleteUserService {
+
+    private final ParameterResolver idamConfig;
+    private final RestClient client;
+
+    public DeleteUserService(RestClient client, ParameterResolver idamConfig) {
+        this.client = client;
+        this.idamConfig = idamConfig;
+    }
+
+    public void deleteUsers(List<String> batchStaleUserIds) {
+        for (String userId: batchStaleUserIds) {
+            deleteUser(userId);
+        }
+    }
+
+    private void deleteUser(String userId) {
+        boolean deletionSucceeded = false;
+        String path = idamConfig.getDeleteUserPath() + userId;
+        for (int i = 0; i < idamConfig.getNumberOfAttemptsToDelete(); i++) {
+            Response response = client.deleteRequest(idamConfig.getIdamHost(), path);
+            if (response.getStatus() == NO_CONTENT.value()) {
+                deletionSucceeded = true;
+                break;
+            } else {
+                log.error("User '{}' deletion failed (attempt #{}): '{}'", userId, i + 1, response.getStatus());
+            }
+        }
+        if (!deletionSucceeded) {
+            String msg = String.format(
+                "User with id '%s' deletion failed after %s attempts",
+                userId,
+                idamConfig.getNumberOfAttemptsToDelete()
```

Comment:
Please remove the retry functionality for now. We need to apply a common retry strategy for all services.